### PR TITLE
macvlan: make master config as optional

### PR DESCRIPTION
**Pull request by s1061123**
To ease of config, make master config as optional parameter.
In default case, default route interface will be used for master.

### CODE REVIEW

File: plugins/main/macvlan/macvlan.go [hunk 1]
```diff
@@ -51,13 +51,33 @@ func init() {
...
 }
 
+func getDefaultRouteInterfaceName() string {
+	routeToDstIP, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return ""
+	}
+
+	for _, v := range routeToDstIP {
+		l, _ := netlink.LinkByIndex(v.LinkIndex)
```
Comment (matthewdupre) [inline plugins/main/macvlan/macvlan.go]: This could be moved inside the if.

File: plugins/main/macvlan/macvlan.go [hunk 2]
```diff
@@ -51,13 +51,33 @@ func init() {
...
+		if v.Dst == nil {
+			return l.Attrs().Name
+		}
+	}
+
+	return ""
```
Comment (matthewdupre) [inline plugins/main/macvlan/macvlan.go]: Would be nice to pass the errors up: the caller gives a generic one: would be nice to know whether there was no default route, or whether netlink gave an error etc.

File: plugins/main/macvlan/macvlan.go (hunk 1 above)
Comment (matthewdupre) [reply plugins/main/macvlan/macvlan.go]: Also nice to capture any error from here.

File: plugins/main/macvlan/macvlan.go (hunk 2 above)
Comment (s1061123) [reply plugins/main/macvlan/macvlan.go]: thank you for your comments. I changed it.

**Comment by s1061123** [comment]
@matthewdupre could you please take a look into it?

**Comment by dcbw** [comment]
Updated LGTM

**Comment by squeed** [comment]
lgtm

**Pull request merged**